cineast-core/src/main/java/org/vitrivr/cineast/core/features/SingleFrameOCRSearch.java

```diff
@@ -0,0 +1,311 @@
+import java.util.ArrayList;
+import java.util.List;
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfFloat;
+import org.opencv.core.MatOfInt;
+import org.opencv.core.MatOfRotatedRect;
+import org.opencv.core.Point;
+import org.opencv.core.RotatedRect;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.dnn.Dnn;
+import org.opencv.dnn.Net;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.utils.Converters;
+
+/**
+ *  OCR is handled by adding fuzziness / levenshtein-distance support to the query if there are no quotes present (as quotes indicate precision)
+ *  This makes sense here since we expect small errors from OCR sources
+ */
+
+public class SingleFrameOCRSearch extends AbstractTextRetriever {
+
+  public static final String OCR_TABLE_NAME = "features_ocr";
+
+  /**
+   * Default constructor for {@link SingleFrameOCRSearch}.
+   */
+  public SingleFrameOCRSearch() {
+    super(OCR_TABLE_NAME);
+  }
+
+  public void processSegment(SegmentContainer shot) {
+    BufferedImage image = shot.getMostRepresentativeFrame().getImage().getBufferedImage();
+    String text;
+
+    SceneTextOCR det = new SceneTextOCR();
+    text = det.textDetector(image);
+    this.writer.write(new SimpleFulltextFeatureDescriptor(shot.getId(), text));
+
+  }
+
+  @Override
+  protected String enrichQueryTerm(String queryTerm) {
+    if (queryTerm.contains("\"")) {
+      return queryTerm;
+    }
+    return queryTerm + "~1";
+  }
+
+
+  // This class is for initializing and configuring Tesseract using Tess4j
+  class TesseractRecognizer {
+    private final Logger LOGGER = Logger.getLogger( AttentionOCRrecognizer.class.getName() );
+
+    public String recognizer(BufferedImage image){
+      Tesseract tesseract = new Tesseract();
+      tesseract.setDatapath("resources/OCRSearch/TesseractData/tessdata/");
+      tesseract.setLanguage("eng");
+      tesseract.setTessVariable("tessedit_char_whitelist", "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890");
+
+      // PageSegMode 6 allows Tesseract to treat the text as a uniform block. Since I am passing a crop of each bounding box this mode would give the best result.
+      tesseract.setPageSegMode(6);
+      // OcrEngineMode 2 makes use of both the LSTM model and the original Tesseract. This gives better result than just using the LSTM model.
+      tesseract.setOcrEngineMode(2);
+      tesseract.setTessVariable("user_defined_dpi", "300");
+      try {
+        String result = tesseract.doOCR(image);
+        return result;
+      } catch (TesseractException e) {
+        LOGGER.log(Level.ALL, e.toString(), e );
+        return null;
+      }
+    }
+  }
+
+  // This class implements a Tensorflow model of AttentionOCR.
+  class AttentionOCRrecognizer{
+
+    private final Logger LOGGER = Logger.getLogger( AttentionOCRrecognizer.class.getName() );
```

Comment:
Are you using the intellij codestyle as described in the repo readme? At least from this line, it looks like you haven't applied code formatting to your changes.

Reply:
Really sorry about all this, I'll communicate with Ralph and will have everything sorted out within a couple of days.

Reply:
No need to apologize, the PR looks great!